Jellyfin.Server/Program.cs

```diff
@@ -0,0 +1,307 @@
+            await createLogger(appPaths);
+            _loggerFactory = new SerilogLoggerFactory();
+            _logger = _loggerFactory.CreateLogger("Main");
+
+            AppDomain.CurrentDomain.UnhandledException += (sender, e) 
+                => _logger.LogCritical((Exception)e.ExceptionObject, "Unhandled Exception");
+
+            _logger.LogInformation("Jellyfin version: {Version}", version);
+
+            EnvironmentInfo environmentInfo = getEnvironmentInfo();
+            ApplicationHost.LogEnvironmentInfo(_logger, appPaths, environmentInfo);
+
+            SQLitePCL.Batteries_V2.Init();
+
+            // Allow all https requests
+            ServicePointManager.ServerCertificateValidationCallback = new RemoteCertificateValidationCallback(delegate { return true; });
+
+            var fileSystem = new ManagedFileSystem(_loggerFactory.CreateLogger("FileSystem"), environmentInfo, null, appPaths.TempDirectory, true);
+
+            using (var appHost = new CoreAppHost(
+                appPaths,
+                _loggerFactory,
+                options,
+                fileSystem,
+                new PowerManagement(),
+                "embyserver-mono_{version}.zip",
+                environmentInfo,
+                new NullImageEncoder(),
+                new SystemEvents(_loggerFactory.CreateLogger("SystemEvents")),
+                new NetworkManager(_loggerFactory.CreateLogger("NetworkManager"), environmentInfo)))
+            {
+                appHost.Init();
+
+                appHost.ImageProcessor.ImageEncoder = GetImageEncoder(_logger, fileSystem, options, () => appHost.HttpClient, appPaths, environmentInfo, appHost.LocalizationManager);
+
+                _logger.LogInformation("Running startup tasks");
+
+                await appHost.RunStartupTasks();
+
+                // TODO: read input for a stop command
+                // Block main thread until shutdown
+                await ApplicationTaskCompletionSource.Task;
+
+                _logger.LogInformation("Disposing app host");
+            }
+
+            if (_restartOnShutdown)
+            {
+                StartNewInstance(options);
+            }
+
+            return 0;
+        }
+
+        private static ServerApplicationPaths createApplicationPaths(StartupOptions options)
+        {
+            string programDataPath;
+            if (options.ContainsOption("-programdata"))
+            {
+                programDataPath = options.GetOption("-programdata");
+            }
+            else
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    programDataPath = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+                }
+                else
+                {
+                    // $XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored.

```

Comment:
I guess this might be incorrect for Mac, but I don't know for sure.

Reply:
This hasn't changed, still the same logic as before.

Reply:
As I said, not everything might be fixed here, but isn't it worth checking and fixing eventually?

Reply:
Okay, let's fix this in a separate PR.